.github/workflows/go.yml

```diff
@@ -14,8 +14,15 @@ jobs:
       matrix:
         os: 
         - ubuntu-latest
-        - macos-latest
+        arch:
+        - amd64
+        - arm
+        - arm64
         go-version: [1.15.3]
+        include:
```

Comment:
I am wondering if it would make sense to instead of having a script to map binary names, to have:

```
include:
        - os: macos-latest
          arch: amd64
          goarch: x86_64
          go-version: 1.15.3
        - os: ubuntu-latest
          arch: amd64
          goarch: x86_64
          go-version: 1.15.3
        - os: ubuntu-latest
          arch: arm
          goarch: arm
          go-version: 1.15.3
        - os: ubuntu-latest
          arch: arm64
          goarch: arm64
          go-version: 1.15.3
```

It's a bit more verbose, but a bit less magic and eventually allows us to then build for windows as well WDYT?
we can also take this as an improvement of course

Reply:
I think you're right.
It's easier to read if you don't mind the verbosity.

If you want to list `goarch` in this list `Map binary name` is technically not necessary anymore. But I'd keep it instead of interpolating the binary name each time.
Looks cleaner to me:
```
- run: ./whalebrew-$(uname -s)-$(uname -m) version
+ run: ./${{ steps.binary.outputs.binary_name }} version
```
---
I'm happy to update the `matrix` according to your snippet. Should I keep the `Map binary name` step or not?

Reply:
I'm happy to keep it as it is so far.
I will merge and release soon!